datafusion/common/src/join_type.rs

```diff
@@ -44,6 +44,8 @@ pub enum JoinType {
     LeftAnti,
     /// Right Anti Join
     RightAnti,
+    /// Left Mark join, used for correlated subqueries EXIST/IN
```

Comment:
I am not familar with what a LeftMarkJoin is -- can we add some additional context here (e.g. maybe based on the PR's description with a link to the ) to help future readers

Reply:
Added more docs in https://github.com/apache/datafusion/pull/13134/commits/00bf6192990ae7a32cc1bcc2b2bd0346e7966296 

Let me know if there is something missing.